### PR TITLE
Adding sts.assume_role() support to resource_sync

### PR DESCRIPTION
**Pull request by andreivmaksimov**
*Issue #, if available:* Adding an ability to assume an IAM role in source and destination AWS accounts 

*Description of changes:* Adding `--src-role-arn` and `--dst-role-arn` parameters. The `--sts-role-arn` has been removed. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: utilities/resource_sync/README.md [hunk 1]
```diff
@@ -21,8 +21,8 @@ You can run this utility in any location where you have Python and the following
 $ sync.py [-h] [--targets TARGETS] [--src-job-names SRC_JOB_NAMES] [--src-database-names SRC_DATABASE_NAMES] [--src-table-names SRC_TABLE_NAMES] [--src-profile SRC_PROFILE] [--src-region SRC_REGION]
                [--src-s3-endpoint-url SRC_S3_ENDPOINT_URL] [--src-sts-endpoint-url SRC_STS_ENDPOINT_URL] [--src-glue-endpoint-url SRC_GLUE_ENDPOINT_URL] [--dst-profile DST_PROFILE]
                [--dst-region DST_REGION] [--dst-s3-endpoint-url DST_S3_ENDPOINT_URL] [--dst-sts-endpoint-url DST_STS_ENDPOINT_URL] [--dst-glue-endpoint-url DST_GLUE_ENDPOINT_URL]
-               [--sts-role-arn STS_ROLE_ARN] [--skip-no-dag-jobs SKIP_NO_DAG_JOBS] [--overwrite-jobs OVERWRITE_JOBS] [--overwrite-databases OVERWRITE_DATABASES] [--overwrite-tables OVERWRITE_TABLES]
-               [--copy-job-script COPY_JOB_SCRIPT] [--config-path CONFIG_PATH] [--skip-errors] [--dryrun] [--skip-prompt] [-v]
+               [--src-role-arn SRC_ROLE_ARN] [--dst-role-arn DST_ROLE_ARN] [--skip-no-dag-jobs SKIP_NO_DAG_JOBS] [--overwrite-jobs OVERWRITE_JOBS] [--overwrite-databases OVERWRITE_DATABASES]
```
Comment (moomindani) [inline utilities/resource_sync/README.md:24]: I would suggest to keep the current parameter `--sts-role-arn`, and to add the two new parameters `--src-role-arn` and `--dst-role-arn`, and use both `--src-role-arn` and `--sts-role-arn` for the same purpose.

Comment (andreivmaksimov) [reply utilities/resource_sync/README.md]: @moomindani , done.

File: utilities/resource_sync/sync.py [hunk 2]
```diff
@@ -93,21 +111,17 @@
...
-    res = src_sts.assume_role(RoleArn=args.sts_role_arn, RoleSessionName='glue-job-sync')
+if args.dst_role_arn is not None or args.sts_role_arn is not None:
+    dst_sts = boto3.client('sts', endpoint_url=args.dst_sts_endpoint_url)
+    role_arn = args.dst_role_arn if args.dst_role_arn is not None else args.sts_role_arn
+    res = dst_sts.assume_role(RoleArn=role_arn, RoleSessionName='glue-job-sync-dst')
     dst_session_args['aws_access_key_id'] = res['Credentials']['AccessKeyId']
     dst_session_args['aws_secret_access_key'] = res['Credentials']['SecretAccessKey']
     dst_session_args['aws_session_token'] = res['Credentials']['SessionToken']
 
-if args.dst_profile is None and args.sts_role_arn is None:
```
Comment (moomindani) [inline utilities/resource_sync/sync.py:108]: Do we want to remove this validation "if all of args.dst_profile/sts_role_arn/dst_role_arn"?

Comment (moomindani) [reply utilities/resource_sync/sync.py:108]: oops I missed that validation. nevermind.

**Review by moomindani — APPROVED** [review]

**Comment by moomindani** [comment]
Merged. Thank you for your contribution.

**Pull request merged**